File: C/007.c

//Find the 10001 prime number.

#include <stdio.h>

int isprime(int n);

int main(void)
{
	int lim = 10001;
	int count = 0;
	int num = 1;
	while (count < lim)
	{
		num++;
		if (isprime(num))
			count++;
	}
	printf("%d\n", num);
}

int isprime(int n)
{
	if (n <= 1)
		return (0);
	if (n == 2)
		return (1);
	if (n % 2 == 0)
		return (0);
	unsigned int i = 3;
	while (i * i <= n)
	{
		if ((n % i) == 0 )
			return (0);
		i++;
	}
	return (1);
}

File: JavaScript/007.js
function isPrime(num) {
  if (num < 2) return false;
  for (let i = 2; i <= Math.sqrt(num); i++) {
	 if (num % i == 0) return false;
  }
  return true;
}

let i = 0;
let count = 0;
while (count < 10001) {
  if (isPrime(i)) {
	 count++;
  }
  i++;
}
console.log(i - 1); 